cmd/cli/kubectl-kyverno/utils/report/report.go

```diff
@@ -89,37 +39,91 @@ func buildPolicyResults(auditWarn bool, engineResponses ...engineapi.EngineRespo
 						UID:        engineResponse.Resource.GetUID(),
 					},
 				},
-				Scored:   true,
+				Scored:   scored,
 				Category: category,
 				Severity: severity,
 			}
-
 			if ruleResponse.Status() == engineapi.RuleStatusSkip {
 				result.Result = policyreportv1alpha2.StatusSkip
 			} else if ruleResponse.Status() == engineapi.RuleStatusError {
 				result.Result = policyreportv1alpha2.StatusError
 			} else if ruleResponse.Status() == engineapi.RuleStatusPass {
 				result.Result = policyreportv1alpha2.StatusPass
 			} else if ruleResponse.Status() == engineapi.RuleStatusFail {
-				if !scored {
-					result.Result = policyreportv1alpha2.StatusWarn
-				} else if auditWarn && engineResponse.GetValidationFailureAction().Audit() {
+				if !scored || (audit && auditWarn) {
 					result.Result = policyreportv1alpha2.StatusWarn
 				} else {
 					result.Result = policyreportv1alpha2.StatusFail
 				}
 			} else {
-				fmt.Println(ruleResponse)
+				result.Result = policyreportv1alpha2.StatusError
 			}
 			if policy.GetType() == engineapi.KyvernoPolicyType {
 				result.Rule = ruleResponse.Name()
 			}
 			result.Message = ruleResponse.Message()
 			result.Source = kyverno.ValueKyvernoApp
-			result.Timestamp = now
-			results[appname] = append(results[appname], result)
+			result.Timestamp = metav1.Timestamp{Seconds: ruleResponse.Stats().Timestamp()}
+			results[policy] = append(results[policy], result)
 		}
 	}
-
 	return results
 }
+
+func ComputePolicyReports(auditWarn bool, engineResponses ...engineapi.EngineResponse) ([]policyreportv1alpha2.ClusterPolicyReport, []policyreportv1alpha2.PolicyReport) {
+	var clustered []policyreportv1alpha2.ClusterPolicyReport
+	var namespaced []policyreportv1alpha2.PolicyReport
+	perPolicyResults := ComputePolicyReportResultsPerPolicy(auditWarn, engineResponses...)
+	for policy, results := range perPolicyResults {
+		if policy.GetNamespace() == "" {
+			report := policyreportv1alpha2.ClusterPolicyReport{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: policyreportv1alpha2.SchemeGroupVersion.String(),
+					Kind:       "ClusterPolicyReport",
+				},
+				Results: results,
+				Summary: reportutils.CalculateSummary(results),
+			}
+			report.SetName(policy.GetName())
+			clustered = append(clustered, report)
+		} else {
+			report := policyreportv1alpha2.PolicyReport{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: policyreportv1alpha2.SchemeGroupVersion.String(),
+					Kind:       "PolicyReport",
+				},
+				Results: results,
+				Summary: reportutils.CalculateSummary(results),
+			}
+			report.SetName(policy.GetName())
+			report.SetNamespace(policy.GetNamespace())
+			namespaced = append(namespaced, report)
+		}
+	}
+	return clustered, namespaced
+}
+
+func MergeClusterReports(clustered []policyreportv1alpha2.ClusterPolicyReport, namespaced []policyreportv1alpha2.PolicyReport) policyreportv1alpha2.ClusterPolicyReport {
```

Comment:
Why everything is merged to clsuterpolicyreports?

Reply:
Not everything, cluster policy reports and policy reports that don't have a namespace.

Reply:
It doesn't make sense to me but couldn't figure out why.
Added a TODO in the code to change that at some point, I think it's because we have another bug somewhere and it should not be necessary anymore when the code is cleaned up.

Reply:
>Added a TODO in the code to change that at some point, I think it's because we have another bug somewhere and it should not be necessary anymore when the code is cleaned up.

Makes sense, we ain't there yet.

Reply:
I think no but... it's the CLI code 🤷